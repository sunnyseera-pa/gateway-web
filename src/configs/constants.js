export const DISPLAY_DATE_STANDARD = 'd MMMM yyyy';
export const TEXTAREA_ROWS = 8;

export const DATASETS_STATUS_ACTIVE = 'active,draft';
export const STATUS_INREVIEW = 'inReview';
export const STATUS_REJECTED = 'rejected';
export const STATUS_ARCHIVE = 'archive';
