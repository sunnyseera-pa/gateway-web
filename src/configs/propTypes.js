import PropTypes from 'prop-types';
import _ from 'lodash';

export const COMMON_PROP_TYPES = {
	ml: PropTypes.number,
	mr: PropTypes.number,
	mb: PropTypes.number,
	mt: PropTypes.number,
};

export const omit = (propTypes, exclude) => {
	return _.omit(propTypes, exclude);
};

export const addCommonPropTypes = propTypes => {
	return {
		...propTypes,
		...COMMON_PROP_TYPES,
	};
};
