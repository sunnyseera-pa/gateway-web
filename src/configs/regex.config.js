export const regExpConfig = {
	httpUrl: /^([http|https]+:\/\/[a-z]+)([^/]*)/,
	cmsUrl: /^([http|https]+:\/\/[a-z]+)\.([^/]*)/,
	alphaNumeric: /^(?=.*[a-zA-Z])(?=.*[0-9])[a-zA-Z0-9]+$/,
	number: /^[0-9]*$/,
};
