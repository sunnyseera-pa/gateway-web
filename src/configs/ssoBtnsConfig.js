import googleIcon from '../images/icons/google-logo.png';
import instIcon from '../images/icons/institution-icon.png';
import microsoftIcon from '../images/icons/microsoft-logo.png';
import linkedInIcon from '../images/icons/LI-In.png';

export const ssoBtnsConfig = [
	{ id: 'google', text: 'Sign in with Google', authURL: '/auth/google', img: googleIcon, active: false },
	{ id: 'openAthens', text: 'Sign in with institution', authURL: '/auth/oidc', img: instIcon, active: false },
	{ id: 'linkedin', text: 'Sign in with LinkedIn', authURL: '/auth/linkedin', img: linkedInIcon, active: false },
	// { id: 'azure', text: 'Sign in with Microsoft', authURL: '/auth/azure', img: microsoftIcon, active: false },
];
