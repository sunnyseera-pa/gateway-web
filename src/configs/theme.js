import isNil from 'lodash/isNil';

export const getSize = (value, { base: { increment, unit } }) => {
	return `${value * increment}${unit}`;
};

export const getSpacingStyle = (prop, value, theme) => {
	return typeof value === 'number' ? `${prop}: ${getSize(value, theme)};` : '';
};

export const getCommonStyle = (prop, value) => {
	return !isNil(value) ? `${prop}: ${value};` : '';
};

export const getCommonStyles = ({ ml, mr, mb, mt, width, maxWidth, minWidth }, theme) => {
	return `
		${getSpacingStyle('margin-left', ml, theme)}
		${getSpacingStyle('margin-right', mr, theme)}
		${getSpacingStyle('margin-bottom', mb, theme)}
		${getSpacingStyle('margin-top', mt, theme)}
		${getCommonStyle('width', width)}
		${getCommonStyle('max-width', maxWidth)}
		${getCommonStyle('min-width', minWidth)}
	`;
};

export const THEME_INPUT = {
	sizes: {
		small: {
			height: '30px',
		},
		default: {
			height: '40px',
		},
		large: {
			height: '50px',
		},
	},
	variants: {
		primary: {
			background: 'white',
			borderColor: 'grey200',
		},
		secondary: {
			background: 'grey100',
			borderColor: 'grey100',
		},
	},
};

export const THEME_FONT_SIZES = {
	xxs: '10px',
	xs: '12px',
	sm: '13px',
	md: '14px',
	default: '14px',
	lg: '16px',
	xl: '20px',
	'2xl': '24px',
	'3xl': '28px',
	'4xl': '32px',
	'5xl': '40px',
};

export const theme = {
	base: {
		increment: 4,
		unit: 'px',
	},
	font: {
		size: {
			default: '14px',
		},
	},
	colors: {
		white: '#fff',
		transparent: 'transparent',
		inherit: 'inherit',
		purple500: '#475DA7',
		green600: '#329276',
		green700: '#2c8267',
		grey: '#F6F7F8',
		grey100: '#F6F7F8',
		grey200: '#EEE',
		grey300: '#E2E2E2',
		grey400: '#D0D3D4',
		grey700: '#53575A',
		grey700Alt: '#848E97',
		grey800: '#3C3C3B',
		red600: '#EF3F4B',
		purple: '#475da7',
		teal: '#3db28c',
	},
	components: {
		Icon: {
			sizes: THEME_FONT_SIZES,
		},
		Input: THEME_INPUT,
		Dropdown: THEME_INPUT,
		Textarea: THEME_INPUT,
		Checkbox: {
			height: '20px',
			width: '20px',
			variants: {
				primary: {
					borderColor: 'grey200',
					backgroundDisabled: 'grey100',
					checkedBackground: 'green700',
					hoverBackground: 'grey200',
				},
				secondary: {
					borderColor: 'grey200',
					backgroundDisabled: 'grey100',
					checkedBackground: 'grey700',
					hoverBackground: 'grey200',
				},
			},
		},
		Typeahead: THEME_INPUT,
	},
};
