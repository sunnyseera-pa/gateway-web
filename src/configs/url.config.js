import { regExpConfig } from './regex.config';

/**
 * [_buildUrl]
 *
 * @param  {urlType[string]}
 * @description Return full URL for either CMS or App
 */
const _buildUrl = (urlType) => {
    // 1. destructure window
    let { location: { href, origin }} = window;
    if (href.includes('appspot.com')) {
        return origin;
    } 
    else if (!href.includes('localhost')) {
        let regArray = _getRegexURL(urlType, href);
        if (regArray) {
            let [one, two, url] = regArray
            //add -api to the sub domain for API requests
            switch(urlType) {
                case 'cms':
                    return `https://${url}`;
                default :
                    return `https://api${url}`;
            }  
        }
    } else {
        return 'http://localhost:3001'
    }
}

/**
 * [_getRegex]
 * 
 * @param {urlType[string], href{[string]}}
 * @description Returns regex test based on url type 
 */
const _getRegexURL = (urlType, href) => {
    console.log(`GET REGEX URL ${urlType}`);
    switch(urlType) {
        case 'cms':
            return regExpConfig.cmsUrl.exec(href);
        default :
            return regExpConfig.httpUrl.exec(href);
    } 
}

export const baseURL = _buildUrl('http');
export const cmsURL = _buildUrl('cms');