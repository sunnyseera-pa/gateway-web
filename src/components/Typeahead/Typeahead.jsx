/** @jsx jsx */
import { cx } from '@emotion/css';
import { jsx } from '@emotion/react';
import PropTypes from 'prop-types';
import { Typeahead as BootstrapTypeahead } from 'react-bootstrap-typeahead';
import 'react-bootstrap-typeahead/css/Typeahead.css';
import { addCommonPropTypes, pick } from '../../configs/propTypes';
import Input from '../Input';
import { PROP_TYPES_INPUT } from '../Input/Input.propTypes';

const Typeahead = ({
	className,
	selected,
	variant,
	textPrepend,
	iconPrepend,
	textAppend,
	iconAppend,
	mt,
	mr,
	mb,
	ml,
	minWidth,
	maxWidth,
	width,
	...outerProps
}) => {
	return (
		<Input
			className={cx(className, 'ui-Typeahead')}
			variant={variant}
			textPrepend={textPrepend}
			iconPrepend={iconPrepend}
			textAppend={textAppend}
			iconAppend={iconAppend}
			mt={mt}
			mb={mb}
			mr={mr}
			ml={ml}
			width={width}
			maxWidth={maxWidth}
			minWidth={minWidth}>
			<BootstrapTypeahead {...outerProps} />
		</Input>
	);
};

Typeahead.propTypes = {
	...pick(PROP_TYPES_INPUT, ['className', 'selected', 'variant', 'textPrepend', 'iconPrepend', 'textAppend', 'iconAppend']),
	selected: PropTypes.array,
};

Typeahead.defaultProps = addCommonPropTypes({
	selected: [],
	variant: 'primary',
});

export default Typeahead;
