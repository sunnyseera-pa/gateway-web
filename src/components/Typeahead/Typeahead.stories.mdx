import { Meta, Story, Canvas } from '@storybook/addon-docs';
import Typeahead from './Typeahead';

<Meta title='Typeahead' component={Typeahead} />

<Canvas>
	<Story name='Basic'>
		{() => {
			const [selected, setSelected] = React.useState();
			const handleChange = values => {
				setSelected(values);
			};
			return <Typeahead options={['United Kingdon,Cambridge', 'United States']} selected={selected} onChange={handleChange} multiple />;
		}}
	</Story>
</Canvas>
