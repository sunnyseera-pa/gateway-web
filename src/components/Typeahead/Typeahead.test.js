import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import React from 'react';
import Typeahead from '.';
import { omit, pick } from '../../configs/propTypes';

const mockInput = jest.fn();
const mockOnChange = jest.fn();
const mockBootstrapTypeahead = jest.fn();

jest.mock('../Input', () => props => {
	mockInput(props);
	return <div>{props.children}</div>;
});

jest.mock('react-bootstrap-typeahead', () => {
	return {
		...jest.requireActual('react-bootstrap-typeahead'),
		Typeahead: props => {
			mockBootstrapTypeahead(props);
			return <div />;
		},
	};
});

const props = {
	options: ['United Kingdon, Cambridge', 'United States'],
	onChange: mockOnChange,
	iconPrepend: 'Icon prepend',
	textPrepend: 'Text prepend',
	iconAppend: 'Icon append',
	textAppend: 'Text append',
	className: 'additional-class',
	mt: 1,
	mb: 2,
	mr: 3,
	ml: 4,
	minWidth: '100',
	maxWidth: '200',
	width: '300',
};

let input;
let wrapper;

describe('Given the Typeahead component', () => {
	describe('When it is rendered without default value', () => {
		beforeAll(() => {
			wrapper = render(<Typeahead {...props} />, {
				wrapper: Providers,
			});

			input = document.querySelectorAll('input')[0];
		});

		it('Then should match the snapshot', () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then should call the input with the correct parameters', () => {
			expect(mockInput.mock.calls[0][0]).toMatchObject({
				...omit(props, ['onChange', 'options']),
				variant: 'primary',
				className: 'additional-class ui-Typeahead',
			});
		});

		it('Then should call the typeahead with the correct parameters', () => {
			expect(mockBootstrapTypeahead).toHaveBeenCalledWith(pick(props, ['onChange', 'options']));
		});
	});
});
