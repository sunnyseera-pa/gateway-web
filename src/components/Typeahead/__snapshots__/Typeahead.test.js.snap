// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Typeahead component When it is rendered without default value Then should match the snapshot 1`] = `
<div>
  <div>
    <div />
  </div>
</div>
`;
